File: statsig-ffi/bindings/java/src/main/java/com/statsig/FeatureGate.java
package com.statsig;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class FeatureGate {
    public String name;
    public boolean value;

    @SerializedName("rule_id")
    public String ruleID;

    @SerializedName("details")
    public EvaluationDetails evaluationDetails;

    @Expose(serialize = false, deserialize = false)
    String rawJson;

    FeatureGate(String name, boolean value, String ruleID, EvaluationDetails evaluationDetails
    ) {
        this.name = name;
        this.value = value;
        this.ruleID = ruleID;
        this.evaluationDetails = evaluationDetails;
    }

    public String getRawJson() {
        return rawJson;
    }

    void setRawJson(String rawJson) {
        this.rawJson = rawJson;
    }
}

File: statsig-dotnet/src/Statsig/FeatureGate.cs
using Newtonsoft.Json.Linq;
using Newtonsoft.Json;

namespace Statsig
{
    public interface IFeatureGate
    {
        string Name { get; }
        bool Value { get; }
        string RuleID { get; }
        string? IDType { get; }
        EvaluationDetails? EvaluationDetails { get; }
    }

    public class FeatureGate : IFeatureGate
    {
        [JsonProperty("name")]
        public string Name { get; }

        [JsonProperty("value")]
        public bool Value { get; }

        [JsonProperty("rule_id")]
        public string RuleID { get; }

        [JsonProperty("id_type")]
        public string? IDType { get; }

        [JsonProperty("details")]
        public EvaluationDetails? EvaluationDetails { get; }

        internal FeatureGate(string rawJson)
        {
            var jsonObject = JObject.Parse(rawJson);
            Name = jsonObject["name"]?.ToString() ?? string.Empty;
            Value = jsonObject["value"]?.ToObject<bool>() ?? false;
            RuleID = jsonObject["rule_id"]?.ToString() ?? string.Empty;
            EvaluationDetails = jsonObject["details"]?.ToObject<EvaluationDetails>();
            IDType = jsonObject["id_type"]?.ToString();
        }
    }
}